cpp/examples/basic/CMakeLists.txt

```diff
@@ -1,16 +1,34 @@
-# Copyright (c) 2020-2023, NVIDIA CORPORATION.
+# Copyright (c) 2020-2024, NVIDIA CORPORATION.
 
 cmake_minimum_required(VERSION 3.26.4)
 
+include(../set_cuda_architecture.cmake)
+
+# initialize cuda architecture
+rapids_cuda_init_architectures(basic_example)
+rapids_cuda_set_architectures(RAPIDS)
+
 project(
   basic_example
   VERSION 0.0.1
   LANGUAGES CXX CUDA
 )
 
 include(../fetch_dependencies.cmake)
-
 # Configure your project here
 add_executable(basic_example src/process_csv.cpp)
 target_link_libraries(basic_example PRIVATE cudf::cudf)
 target_compile_features(basic_example PRIVATE cxx_std_17)
+
+# Install the example only if CMAKE_INSTALL_PREFIX is set
+if(NOT $ENV{CMAKE_INSTALL_PREFIX} STREQUAL "")
```

Comment:
This seems like surprising behavior to install if an environment variable is present, especially since this isn't a RAPIDS-specific environment variable. We shouldn't need to set this variable explicitly, since CMake is supposed to read it from the environment. https://cmake.org/cmake/help/latest/variable/CMAKE_INSTALL_PREFIX.html

What happens if you remove this conditional? Do we need to specify something like `install(... COMPONENT examples EXCLUDE_FROM_ALL)`?

See also: https://discourse.cmake.org/t/understanding-the-cmake-component-keyword-in-the-install-command/971

Reply:
Agreed with this. We need a variable (currently I am using the CMAKE_INSTALL_PREFIX itself to test out the CMake's shenanigans) to control installation and have it to set `CMAKE_INSTALL_PREFIX` too in the initial `CMake` command. If not set at this stage, it has to be forced before `install` in CMakeLists.txt using some check